pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go

```diff
@@ -74,31 +74,36 @@ func NewRequestedToCapacityRatio(plArgs *runtime.Unknown, handle framework.Frame
 	}
 
 	resourceToWeightMap := make(resourceToWeightMap)
-	for _, resource := range args.Resources {
+	for _, resource := range pl.Resources {
 		resourceToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
 		if resource.Weight == 0 {
 			// Apply the default weight.
 			resourceToWeightMap[v1.ResourceName(resource.Name)] = 1
 		}
 	}
-	if len(args.Resources) == 0 {
+	if len(pl.Resources) == 0 {
 		// If no resources specified, used the default set.
 		resourceToWeightMap = defaultRequestedRatioResources
 	}
 
-	return &RequestedToCapacityRatio{
-		handle: handle,
-		resourceAllocationScorer: resourceAllocationScorer{
-			RequestedToCapacityRatioName,
-			buildRequestedToCapacityRatioScorerFunction(shape, resourceToWeightMap),
-			resourceToWeightMap,
-		},
-	}, nil
+	pl.resourceAllocationScorer = resourceAllocationScorer{
+		RequestedToCapacityRatioName,
+		buildRequestedToCapacityRatioScorerFunction(shape, resourceToWeightMap),
+		resourceToWeightMap,
+	}
+
+	return pl, nil
+}
+
+// BuildArgs returns the args that were used to build the plugin.
+func (pl *RequestedToCapacityRatio) BuildArgs() interface{} {
+	return pl.RequestedToCapacityRatioArguments
 }
 
 // RequestedToCapacityRatio is a score plugin that allow users to apply bin packing
 // on core resources like CPU, Memory as well as extended resources like accelerators.
 type RequestedToCapacityRatio struct {
+	config.RequestedToCapacityRatioArguments
```

Comment:
Use the type `RequestedToCapacityRatioArgs` instead, making sure to add json annotations to it.